### DISCUSSION TITLE
Extract  Bayesian posterior predicted outputs for all subjects in an excel/csv

### DISCUSSION THREAD

**Discussion by ChrisKonig** [LAPKB/Pmetrics#218] [General]
Hi all, 
I am looking into a method to generate an excel or csv doc for the bayesian post predicted outputs for all subjects on an individual basis. Appreciate any help.

**Comment by mnneely** [comment]
Tidyverse can help you here.  As an example...

```
library(tidyverse) #ensures you have necessary functions loaded

run1 <- PM_load(1) #assumes you have a completed run 1 in the Runs folder
run1$op$data %>% 
filter(pred.type == "post", icen == "median") %>% #filters to posterior predictions based on median of the Bayesian posterior
select(1:4) %>% #select the first 4 columns: id, time, obs, pred
write_csv(file = "myfile.csv") #write the object to a file
```

The above code assumes only one output equation. You'd have to amend your `filter` and `select` if you had more than one output. If you wanted the posterior predictions at frequent (default 5 per hour) intervals plus at the time of observations, use this code:

```
run1 <- PM_load(1) 
run1$post$data %>% 
filter(icen == "median") %>% 
select(id, time, pred) %>% 
write_csv(file = "myfile.csv") #write the object to a file
```

The filtering and selecting are slightly different since the $op and $post data objects are also slightly different.
